### ISSUE TITLE
Run coverageReport failed when not set managedDevices

### ISSUE THREAD

**Issue by LinkDX** [NeoTech-Software/Android-Root-Coverage-Plugin#90]
On release 1.7.0, following check will cause test failed when not set `testOptions.managedDevices` in project's `build.gradle`. `allDevices*AndroidTest` and `connected*AndroidTest` will not be generated when not set this option.

RootCoveragePlugin.kt
```
} else {
    // If this plugin should not run instrumented tests on it's own, at least make sure it runs after those tasks (if they are
    // selected to run as well).
    mustRunAfter("$path:allDevices${name}AndroidTest")
    mustRunAfter("$path:connected${name}AndroidTest")
}
```

Maybe we should remove this check or do this check when related tasks exist?

**Comment by Rolf-Smit** [comment]
Can you be a bit more specific in the error description?

What exactly is going wrong, do you have an exception, stack trace? Missing coverage?

Please share your configuration and how your are attempting to run tests and collect coverage.

**Comment by LinkDX** [comment]
I will get following error message when run `module:coverageReport` :

```
Could not determine the dependencies of task ':module:coverageReport'.
> Task with path ':module:allDevicesDebugAndroidTest' not found in project ':module'.
```

`allDevicesDebugAndroidTest` will be available when I set following setting in module's `build.gradle`

```
android {
    ...
    testOptions {
        managedDevices {
            devices {
                pixel2api30(com.android.build.api.dsl.ManagedVirtualDevice) {
                    device = "Pixel 2"
                }
            }
        }
    }
}
```

When I fork plugin and remove `mustRunAfter` check, it works normally.


Here the setting in my root `build.gradle`:
```
rootCoverage {
    // Overrides the default build variant for specific modules.
    buildVariantOverrides ":module" : "debug"

    // Since 1.4: Sets jacoco.includeNoLocationClasses, so you don't have to. Helpful when using Robolectric
    // which usually requires this attribute to be true
    includeNoLocationClasses true

    executeAndroidTests false
}
```

**Comment by Rolf-Smit** [comment]
Turns out that the `"$path:allDevices${name}AndroidTest"` task only exists if a Gradle Managed Device has been configured.

So to fix this I should check for the existence of this task in the graph before telling the coverage task to always run after that task (task order is important for things such as code coverage, you cannot calculate coverage before tests have executed).

Just removing the `mustRunAfter` statements is thus dangerous as it can cause coverage to go missing if the tasks are not ordered correctly, hence that is not a solution.

TLDR: Working on it.

**Comment by LinkDX** [comment]
Thank you for promptly fixing this issue. :)

**Comment by Rolf-Smit** [comment]
A new 1.8.0-SNAPSHOT build is available where this is fixed (also fixed on release 1.7 with hotfix 1.7.1)
